extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -1,16 +1,65 @@
 package io.quarkus.jackson.deployment;
 
+import java.util.Collection;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.*;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
 
 public class JacksonProcessor {
 
+    private static final DotName JSON_DESERIALIZE = DotName.createSimple(JsonDeserialize.class.getName());
+    private static final DotName BUILDER_VOID = DotName.createSimple(Void.class.getName());
+
+    private static final String JSON_DESERIALIZE_FIELD_BUILDER = "builder";
+
+    @Inject
+    BuildProducer<ReflectiveClassBuildItem> reflectiveClass;
+
+    @Inject
+    CombinedIndexBuildItem combinedIndexBuildItem;
+
     @BuildStep
-    void register(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false,
+    void register() {
+        addReflectiveClass(true, false,
                 "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector",
-                "com.fasterxml.jackson.databind.ser.std.SqlDateSerializer"));
+                "com.fasterxml.jackson.databind.ser.std.SqlDateSerializer");
+
+        IndexView index = combinedIndexBuildItem.getIndex();
+
+        Collection<AnnotationInstance> pojoBuilderInstances = index.getAnnotations(JSON_DESERIALIZE);
+        for (AnnotationInstance pojoBuilderInstance : pojoBuilderInstances) {
+            addReflectiveClassInheritanceTree(pojoBuilderInstance.target().asClass());
+
+            AnnotationValue annotationValue = pojoBuilderInstance.value(JSON_DESERIALIZE_FIELD_BUILDER);
+            if (null != annotationValue && AnnotationValue.Kind.CLASS.equals(annotationValue.kind())) {
+                DotName builderClassName = annotationValue.asClass().name();
+                if (!BUILDER_VOID.equals(builderClassName)) {
+                    ClassInfo builderClass = combinedIndexBuildItem.getIndex().getClassByName(builderClassName);
+                    addReflectiveClassInheritanceTree(builderClass);
+                }
+            }
+        }
     }
 
+    private void addReflectiveClassInheritanceTree(ClassInfo classInfo) {
```

Comment:
Have you explored the use of `ReflectiveHierarchyBuildItem` instead of doing this? I think it would be a very good fit for this use case